src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -229,6 +230,7 @@ class VisitService(
     )
   }
 
+  @Transactional(propagation = NEVER)
```

Comment:
Are you sure this means "don't start a transaction for this method"? Spring documentation suggests it means something different - but I'm not sure since I have never used it before?

(obviously you could remove the class annotation and add one to each method except this one)

Reply:
My understand of NEVER is that it checks to see if there is an existing transaction and if so throws an Illegal state exception. So by adding this to the method I'm preventing the use of this method within an existing transaction,.. Integration test would fail. The call to persistVisitHelper then creates a new transaction.